src/Config/Converters/Utf8JsonReaderExtensions.cs

```diff
@@ -41,4 +42,115 @@ static internal class Utf8JsonReaderExtensions
+    /// </summary>
+    /// <param name="element">The JSON element to process.</param>
+    /// <param name="writer">The JSON writer to write the result to.</param>
+    private static void ReplaceEnvVarsAndWrite(JsonElement element, Utf8JsonWriter writer)
+    {
+        switch (element.ValueKind)
+        {
+            case JsonValueKind.Object:
+                writer.WriteStartObject();
+                foreach (JsonProperty property in element.EnumerateObject())
+                {
+                    writer.WritePropertyName(property.Name);
+                    // Recursively process each property of the object
+                    ReplaceEnvVarsAndWrite(property.Value, writer);
+                }
+
+                writer.WriteEndObject();
+                break;
+            case JsonValueKind.Array:
+                writer.WriteStartArray();
+                foreach (JsonElement item in element.EnumerateArray())
+                {
+                    // Recursively process each item of the array
+                    ReplaceEnvVarsAndWrite(item, writer);
+                }
+
+                writer.WriteEndArray();
+                break;
+            case JsonValueKind.String:
+                string? value = element.GetString();
+                if (value is not null && value.StartsWith("@env('") && value.EndsWith("')"))
+                {
+                    string envVar = value[6..^2];
+
+                    // Check if the environment variable is set
+                    if (!Environment.GetEnvironmentVariables().Contains(envVar))
+                    {
+                        throw new ArgumentException($"Environment variable '{envVar}' is not set.");
+                    }
+
+                    string? envValue = Environment.GetEnvironmentVariable(envVar);
+
+                    // Write the value of the environment variable to the JSON writer
+                    if (envValue == "null")
+                    {
+                        writer.WriteNullValue();
+                    }
+                    else if (bool.TryParse(envValue, out bool boolValue))
+                    {
+                        writer.WriteBooleanValue(boolValue);
+                    }
+                    else if (int.TryParse(envValue, out int intValue))
+                    {
+                        writer.WriteNumberValue(intValue);
+                    }
+                    else if (double.TryParse(envValue, out double doubleValue))
+                    {
+                        writer.WriteNumberValue(doubleValue);
+                    }
+                    else
+                    {
+                        writer.WriteStringValue(envValue);
+                    }
+                }
+                else
+                {
+                    writer.WriteStringValue(value);
+                }
+
+                break;
+            default:
+                // Write the JSON element to the writer as is
```

Comment:
So if the env('<match>') doesnt start the string we end up here correct, but then we don't do any replacements?